mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAnnotationMapper.java

```diff
@@ -66,10 +68,26 @@ public void insertAnnotation(MailboxAnnotation mailboxAnnotation) {
             .value(CassandraAnnotationTable.VALUE, mailboxAnnotation.getValue().get()));
     }
 
+    private List<MailboxAnnotation> getResult(ResultSet resultSet) {
+        return CassandraUtils.convertToStream(resultSet)
+                .map(this::toAnnotation)
+                .collect(Collectors.toList());
+    }
+
+    private MailboxAnnotation toAnnotation(Row row) {
+        return MailboxAnnotation.newInstance(row.getString(CassandraAnnotationTable.KEY), row.getString(CassandraAnnotationTable.VALUE));
+    }
+
     private ResultSet getStoredAnnotationRow() {
-        return session.execute(select(CassandraAnnotationTable.KEY, CassandraAnnotationTable.VALUE)
+        return session.execute(select(CassandraAnnotationTable.FIELDS)
             .from(CassandraAnnotationTable.TABLE_NAME)
             .where(eq(CassandraAnnotationTable.MAILBOX_ID, mailboxId.asUuid())));
     }
 
+    private ResultSet getStoredAnnotationRow(List<String> keys) {
+        return session.execute(select(CassandraAnnotationTable.FIELDS)
+            .from(CassandraAnnotationTable.TABLE_NAME)
+            .where(eq(CassandraAnnotationTable.MAILBOX_ID, mailboxId.asUuid()))
+            .and(in(CassandraAnnotationTable.KEY, keys)));
```

Comment:
Queries compose very well, you can have a common part with getStoredAnnotationRow then apply the filter in this method only.